app/controllers/application_controller.rb

```diff
@@ -5,6 +5,155 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
+  get "/" do
+    erb :index
+  end
+
+  get "/signup" do
+    if logged_in?
+      redirect "/tweets"
+    else
+        erb :"/users/create_user"
+    end
+  end
+
+  post "/signup" do
+    user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+    if user.save
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+  get "/tweets" do
+    if logged_in?
+      @user = User.find(session[:user_id])
+      @tweets = Tweet.all
+      # binding.pry
+      erb :"/tweets/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/login" do
+     if logged_in?
+      redirect "/tweets"
+    else
+      erb :"/users/login"
+    end
+  end
+
+  post "/login" do
+    user = User.find_by(:username => params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get "/logout" do
+     if logged_in?
+         session.clear
+         redirect "/login"
+    else
+      redirect "/"
+    end
+  end
+
+  get '/users/:slug' do
+      @user = User.find(session[:user_id])
+      @slug = params[:slug]
+      @show_user = User.find_by_slug(@slug)
+      erb :'/users/show_user'
+  end
+
+  get '/tweets/new' do
+     if logged_in?
+         erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    @user = User.find(session[:user_id])
+    if !params["content"].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.